tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -1892,6 +1898,8 @@ bool MayPreventVectorization(const HloInstruction& hlo) {
       default:
         return false;
     }
+  } else if (hlo.opcode() == HloOpcode::kReduce) {
+    return false;
```

Comment:
Do we need to be specific about what reduction is vectorized?

Reply:
I suppose you mean to check what inside the to_apply() HloComputation of the reduction.
In theory yes. But it wasn't done before. I'm not sure it is worth doing it.
So I'll add a todo about this to let people know about this.

Reply:
Maybe let's make it more specific: state what kind of reductions still prevent vectorization.

Reply:
> Reduction doesn't prevent vectorization. This is the point

I understand, I mean: could you specify in the comment explicitly what needs to be checked and why, e.g. "return false only for row reductions which are ..."?

Reply:
This function is independent of the reduction being a row or a columns reductions. I updated it to be tell what to check.